### PR TITLE
Make and use locking decorator

### PR DESCRIPTION
**Pull request by penguinland**
Closes https://github.com/keyme/multevents/issues/7

I recommend viewing [this PR with whitespace changes ignored](https://github.com/keyme/multevents/pull/8/files?w=1). 

Tried at my desk: AllEvent and InverseEvent work normally. I haven't tried out the rest, but these changes are formulaic enough that if anything works, everything should. We should write automated tests soon...

### CODE REVIEW

File: events.py [hunk 1]
```diff
@@ -9,6 +10,17 @@ class UsageError(Exception):
     pass
 
 
+def atomic(func):
+    """
+    A decorator to acquire an object's lock for the entirety of a function
```
Comment (cviveiro) [inline events.py]: period at the end?

**Comment by penguinland** [comment]
`self._lock` is now mentioned in only 3 places:
- where it is created/defined
- the `atomic` decorator
- within `ComboEvent.__init__` (which can't easily work with `@atomic` because the lock is not yet created at the start of the function).

**Review by cviveiro — COMMENTED** [review]
Cool. LGTM.

**Comment by penguinland** [comment]
...wait, I should start the name with an underscore because this thing probably shouldn't be considered public.

**Comment by penguinland** [comment]
Period and underscores added.

**Comment by cviveiro** [comment]
Still LGTM

**Pull request merged**